accepted/00075-live-patching-enhancements.md

```diff
@@ -0,0 +1,170 @@
+- Feature Name: Live Patching Enhancements
+- Start Date: (fill with today's date, YYYY-MM-DD)
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Enhance live patching integration by providing EOL information on installed patches.
+
+Improve live patching workflow using CLM.
+
+# Motivation
+[motivation]: #motivation
+
+SUSE only provides live patches for kernels up to 1 year after original release date of that kernel. It means that after one year, the user must reboot and use a new kernel so that they can keep getting live patches for another year.
+
+This feature is about enhancing the live patching experience in SUSE Manager by implementing the following improvements:
+
+1. Add information about the EOL of kernels in the UI
+2. Warn user when a live kernel is approaching EOL, or past EOL
+3. Add CLM filters to make package selection easier
+
+The existing live patching integration in Uyuni is described in the initial [Live Patching RFC](https://github.com/uyuni-project/uyuni-rfc/blob/master/accepted/00027-sle-live-patching.md).
+
+# Detailed design
+[design]: #detailed-design
+
+This design provides solutions for two separate problems regarding to kernel live patching:
+
+1. [Live patching EOL information](#live-patching-eol-information)
+2. [Live patching CLM workflow](#live-patching-clm-workflow)
+
+## 1. Live patching EOL information
+
+SUSE provides lifecycle information on any product to users via 'zypper lifecycle' plugin. The actual data about a product is provided in a plain CSV file in a supplementary package per product. In case of live patching, this package is called `lifecycle-data-sle-module-live-patching`. Zypper lifecycle plugin reads this data to display human-friendly information to users.
+
+Uyuni shall use this information to provide the following:
+ - Display EOL dates for the current live patch in system details
+ - Display an out-of-support systems list in the home page
+ - Use the notifications system and/or send emails to report live patched systems approaching/past EOL
+
+### Data Retrieval
+
+This section evaluates three alternative implementations for this feature.
+
+#### Call 'zypper lifecycle' on client
+
+Uyuni calls `zypper lifecycle` on the client and collects the data via Salt during package profile update. The data is then stored in the database per client along with the kernel live version data.
+
+**Pros:**
+ - `zypper lifecycle` is publicly supported
+ - Can be attached to profile update data with an existing process
```

Comment:
> Does that imply that correct channel assignment should be there for the Server to be able to know correct EOL data? What would happen if critical channels are missing for example?

`zypper lifecycle` seems to work independently from the assigned channels, so as long as the live patching product (the release package) is installed, it will report correctly.

Reply:
> `zypper lifecycle` seems to work independently from the assigned channels, so as long as the live patching product (the release package) is installed, it will report correctly.

Depends. You need to install the package to get the data. After the package is installed you can also run this without the channel assigned. The problem is, that you will not get package updates and therefore no update of the lifecylce data when the channel is not assigned.
As sometimes packages will be limited later as it depends on upstream supportability you may think that you have support, but you just have no up to date data.
For Livepatching Lifecycle data this should not happen as long as you update the lifecycle package with the kernel lifepatch you should be fine. And AFAIK both comes from the same channel.